src/backend/commands/tablespace.c

```diff
@@ -1532,6 +1534,18 @@ check_temp_tablespaces(char **newval, void **extra, GucSource source)
 	char	   *rawname;
 	List	   *namelist;
 
+	/* If encountered fallback, return empty extra*/
```

Comment:
Iirc, this function is used for both `temp_spill_files_tablespaces` and `temp_tablespaces`. Is it possible to take this branch only for spill files?
```suggestion
	/* If a fallback string is encountered, set numSpcs = 0 to use
     * temp_tablespaces instead. */
```

Reply:
Not really, there is no way to determine which GUC we're checking without duplicating this whole function. However this probably isn't an issue since fallback value would just act like `''` for temp_tablespaces.

Reply:
I guess it's OK as long it's not blowing anything else up. Can anything go wrong if user manually sets one of the GUCs to `__EMPTY__`?

Reply:
No, it's equivalent to resetting the GUC (for both of them). Also, `__EMPTY__` is still quoted and cannot be `SET` correctly, changed it to `__empty__`. (It seems like for the tests to pass any non-existent tablespace would work, that's why they passed before)